development/workflow.rst

```diff
@@ -0,0 +1,501 @@
+
+Next, create this branch on the GitHub remote.
+From a package's repository in ``lsstsw/build``:
+
+.. prompt:: bash
+
+   git push -u
+
+*(repeat for other packages in development)*
+
+This initial push will create a remote branch ``origin/tickets/DM-MMMM`` and *track* it so that you can simply ``git push`` and ``git pull`` without arguments between the ticket branch on the ``origin`` remote and your local clone.
+
+.. _lsstsw-workflow-working-ticket-declare:
+
+Step 2. Declare these Repositories to EUPS
+------------------------------------------
+
+We need to tell EUPS about these development repositories (with ``eups declare``) and set them up for building (with ``setup``).
+From a package's repository in ``lsstsw/build``:
+
+.. prompt:: bash
+
+   eups declare -r . -t $USER {{package_name}} git
+   setup -r . -t $USER
+
+*(repeat for other packages in development)*
+
+Unpacking the ``eups declare`` arguments:
+
+- ``-r .`` is the path to the package's repository, which is the current working directory.
+  You don't *need* to be in the repository's directory if you provide the path appropriately.
+- ``-t $USER`` sets the EUPS *tag*.
+  We use this because your username (``$USER``) is an allowed EUPS tag.
+- ``git`` is used as an EUPS *version*.
+  Semantically we default to calling the version "``git``" to indicate this package's version is the HEAD of a git development branch.
+
+In the above ``eups declare`` command we associated the package version "``git``" with the tag "``$USER``."
+Then when we ``setup`` the package, we told EUPS to setup the package and its dependencies with the version associated with the ``$USER`` tag.
+If the ``$USER`` tag isn't found for dependencies, EUPS will revert to using versions of dependencies linked to the ``current`` tag.
+This is why we initially declared the entire lsstsw repository to have the version ``current``.
+
+.. why not setup -j? Means setup *just* this package, no dependencies
+
+.. _lsstsw-workflow-working-ticket-scons:
+
+Step 3. Compile and Test with SCons
+-----------------------------------
+
+Develop the package(s) as you normally would.
+To build the Stack with the newly-developed package, run SCons from the repository of a package being developed:
+
+.. prompt:: bash
+
+   scons -Q -j 6 opt=3 
+
+These flags tell SCons to build with flags:
+
+- ``-Q``: reduce logging to the terminal,
+- ``-j 6``: build in parallel (e.g., with '6' CPUs),
+- ``opt=3``: build with level 3 optimization.
+
+This ``scons`` command will run several targets by default, in sequence:
+
+1. ``lib``: build the C++ code and SWIG interface layer
+2. ``python``: install the Python code
+3. ``tests``: run the test suite
+4. ``example``: compile and run the examples, and
+5. ``doc``: compile Doxygen-based documentation.
+
+You can build a subset of these targets by specifying one explicitly.
+To simply compile C++, SWIG, build the Python package and run tests:
+
+.. prompt:: bash
+
+   scons -q -j 6 opt=3 tests
+
```

Comment:
If you're building multiple packages that interact, you need to use "rebuild -r {branch}", I believe.